sdk/python/foundation-models/system/inference/text-generation/falcon-safe-online-deployment.ipynb

```diff
@@ -0,0 +1,617 @@
+    "uai_client_id = \"\"\n",
+    "if uai_name != \"\":\n",
+    "    from azure.mgmt.msi import ManagedServiceIdentityClient\n",
+    "    from azure.mgmt.msi.models import Identity\n",
+    "\n",
+    "    msi_client = ManagedServiceIdentityClient(\n",
+    "        subscription_id=subscription_id,\n",
+    "        credential=credential,\n",
+    "    )\n",
+    "    uai_resource = msi_client.user_assigned_identities.get(resource_group, uai_name)\n",
+    "    uai_id = uai_resource.id\n",
+    "    uai_client_id = uai_resource.client_id"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 3.3 Create Falcon online endpoint\n",
+    "This step may take a few minutes."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Create endpoint"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azure.ai.ml.entities import (\n",
+    "    ManagedOnlineEndpoint,\n",
+    "    IdentityConfiguration,\n",
+    "    ManagedIdentityConfiguration,\n",
+    ")\n",
+    "\n",
+    "# Check if the endpoint already exists in the workspace\n",
+    "try:\n",
+    "    endpoint = ml_client.online_endpoints.get(endpoint_name)\n",
+    "    print(\"---Endpoint already exists---\")\n",
+    "except:\n",
+    "    # Create an online endpoint if it doesn't exist\n",
+    "\n",
+    "    # Define the endpoint\n",
+    "    endpoint = ManagedOnlineEndpoint(\n",
+    "        name=endpoint_name,\n",
+    "        description=\"Test endpoint for model\",\n",
+    "        identity=IdentityConfiguration(\n",
+    "            type=\"user_assigned\",\n",
+    "            user_assigned_identities=[ManagedIdentityConfiguration(resource_id=uai_id)],\n",
+    "        )\n",
+    "        if uai_id != \"\"\n",
+    "        else None,\n",
+    "    )\n",
+    "\n",
+    "    # Trigger the endpoint creation\n",
+    "    try:\n",
+    "        ml_client.begin_create_or_update(endpoint).wait()\n",
+    "        print(\"\\n---Endpoint created successfully---\\n\")\n",
+    "    except Exception as err:\n",
+    "        raise RuntimeError(\n",
+    "            f\"Endpoint creation failed. Detailed Response:\\n{err}\"\n",
+    "        ) from err"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### 3.4 Deploy Falcon model"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Create deployment"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Initialize deployment parameters"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "REQUEST_TIMEOUT_MS = 90000\n",
+    "CODE = \"./llama-files/score/default\"\n",
```

Comment:
should this be llama?

Reply:
It is identical, so I reuse llama file here.